win32ss/drivers/font/ftfd/rosglue.c

```diff
@@ -36,52 +36,63 @@ DbgPrint(IN PCCH Format, IN ...)
  * buffer (need to copy the old contents to the new buffer). So, allocate
  * extra space for a size_t, store the allocated size in there and return
  * the address just past it as the allocated buffer.
+ * On win64 we need to align the allocation to 16 bytes, otherwise 8 bytes.
  */
+typedef struct _MALLOC_HEADER
+{
+    SIZE_T Size;
+    SIZE_T Alignment;
+} MALLOC_HEADER, * PMALLOC_HEADER;
 
 void *
 malloc(size_t Size)
 {
-    void *Object;
+    PMALLOC_HEADER Header;
 
-    Object = EngAllocMem(0, sizeof(size_t) + Size, TAG_FREETYPE);
-    if (Object != NULL)
+    Header = EngAllocMem(0, sizeof(MALLOC_HEADER) + Size, TAG_FREETYPE);
+    if (Header == NULL)
     {
-        *((size_t *)Object) = Size;
-        Object = (void *)((size_t *)Object + 1);
+        return NULL;
```

Comment:
https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/malloc?view=msvc-170
> malloc sets errno to ENOMEM if a memory allocation fails or if the amount of memory requested exceeds _HEAP_MAXREQ.

Reply:
errno is not needed here.